cni/test/e2e-taint/e2e_taint_test.go

```diff
@@ -0,0 +1,256 @@
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"istio.io/istio/pkg/test/kube"
+
+	v12 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"istio.io/istio/cni/pkg/taint"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+const (
+	TimeDuration = 60 * time.Second
+	TimeDelay    = 10 * time.Millisecond
+)
+
+var (
+	Test1NS = v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test1",
+		},
+	}
+	Test2NS = v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test2",
+		},
+	}
+)
+
+// Set up Kubernetes client using kubeconfig (or in-cluster config if no file provided)
+func clientSetup() (clientset *client.Clientset, err error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	config, err := kubeConfig.ClientConfig()
+	if err != nil {
+		return
+	}
+	clientset, err = client.NewForConfig(config)
+	return
+}
+
+//read file given location
+func buildDaemonSetFromYAML(fileLocation string) (obj v12.DaemonSet, err error) {
+	reader, err := os.Open(fileLocation)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	obj = v12.DaemonSet{}
+	err = yaml.NewYAMLOrJSONDecoder(reader, 4096).Decode(&obj)
+	return
+}
+func createDaemonset(client *client.Interface, fileLocation string) (daemonset *v12.DaemonSet, err error) {
+	obj, err := buildDaemonSetFromYAML(fileLocation)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	daemonset, err = (*client).AppsV1().DaemonSets(obj.Namespace).Create(context.TODO(), &obj, metav1.CreateOptions{})
+	if err != nil {
+		log.Fatalf(err.Error())
+		return nil, err
+	}
+	return daemonset, err
+}
+
+//build configmap given name and namespace in cmd.ControllerOptions and location
+func buildConfigIfNotExists(clientSet client.Interface, options taint.Options, fileLocation string) (configmap *v1.ConfigMap, err error) {
+	err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Delete(context.TODO(), options.ConfigmapName, metav1.DeleteOptions{})
+	fmt.Println(filepath.Abs("./"))
+	currpath, err := filepath.Abs("./")
+	cmd := exec.Command("kubectl", "create", "configmap", options.ConfigmapName, "-n", options.ConfigmapNamespace, "--from-file="+fmt.Sprintf("%s/%s/", currpath, fileLocation))
+	fmt.Println(cmd.String())
+	err = cmd.Run()
+	err = retry.UntilSuccess(func() error {
+		configmap, err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Get(context.TODO(), options.ConfigmapName, metav1.GetOptions{})
+		return err
+	}, retry.Converge(1), retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	configmap, err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Get(context.TODO(), options.ConfigmapName, metav1.GetOptions{})
```

Comment:
we don't need this, already have retry.UntilSuccess getting it above?

Reply:
removed this line